Origin: OpenSuSE, https://build.opensuse.org/stage/package/view_file?file=pcmanfm-rename-to-pcmanfm.patch&package=pcmanfm&project=X11%3Alxde
Reviewed-By: Andrea Florio
Last-Update: 2010-05-10

--- pcmanfm-0.9.5.orig/configure.ac
+++ pcmanfm-0.9.5/configure.ac
@@ -93,7 +93,7 @@ PKG_CHECK_MODULES(MENU_CACHE, [$menu_mod
 AC_SUBST(MENU_CACHE_CFLAGS)
 AC_SUBST(MENU_CACHE_LIBS)
 
-GETTEXT_PACKAGE=pcmanfm2
+GETTEXT_PACKAGE=pcmanfm
 AC_SUBST(GETTEXT_PACKAGE)
 
 # Generate po/LINGUAS on the fly rather than relying on translators
@@ -116,7 +116,7 @@ AC_CONFIG_FILES([
     data/Makefile
     data/ui/Makefile
     data/ui/about.glade
-    data/pcmanfm2.desktop
+    data/pcmanfm.desktop
 ])
 
 AC_OUTPUT
--- pcmanfm-0.9.5.orig/po/POTFILES.skip
+++ pcmanfm-0.9.5/po/POTFILES.skip
@@ -1,3 +1,3 @@
-data/pcmanfm2.desktop.in
+data/pcmanfm.desktop.in
 #we transtale .in file
 data/ui/about.glade
--- /dev/null
+++ pcmanfm-0.9.5/data/pcmanfm.desktop.in
@@ -0,0 +1,28 @@
+[Desktop Entry]
+Type=Application
+Icon=system-file-manager
+Name=File Manager
+Name[de]=Dateimanager
+Name[et]=Failihaldur
+Name[fi]=Tiedostonhallinta
+Name[ja]=ファイルマネージャ
+Name[ru]=Файловый менеджер
+Name[zh_TW]=檔案管理程式
+GenericName=File Manager
+GenericName[de]=Dateimanager
+GenericName[et]=Failihaldur
+GenericName[fi]=Tiedostonhallinta
+GenericName[ja]=ファイルマネージャ
+GenericName[ru]=Файловый менеджер
+GenericName[zh_TW]=檔案管理程式
+Comment=Browse the file system and manage the files
+Comment[de]=Das Dateisystem durchsuchen und Dateien verwalten
+Comment[et]=Lehitse failisüsteemi ja halda faile
+Comment[fi]=Selaa tiedostojärjestelmää ja hallitse tiedostoja
+Comment[ja]=ファイルシステムをブラウズし、ファイルの管理を行います
+Comment[zh_TW]=瀏覽檔案系統及管理檔案
+Categories=FileManager;Utility;Core;GTK;
+Exec=pcmanfm %U
+StartupNotify=true
+Terminal=false
+MimeType=x-directory/normal;inode/directory;
--- pcmanfm-0.9.5.orig/data/Makefile.am
+++ pcmanfm-0.9.5/data/Makefile.am
@@ -4,5 +4,5 @@ SUBDIRS=ui
 
 desktopdir=$(datadir)/applications
 desktop_DATA= \
-	pcmanfm2.desktop \
+	pcmanfm.desktop \
 	$(NULL)
--- pcmanfm-0.9.5.orig/data/ui/Makefile.am
+++ pcmanfm-0.9.5/data/ui/Makefile.am
@@ -1,7 +1,7 @@
 NULL=
 
 # GtkBuilder UI definition files
-uidir=$(datadir)/pcmanfm2/ui
+uidir=$(datadir)/pcmanfm/ui
 ui_in_files= \
 	about.glade \
 	pref.glade \
--- pcmanfm-0.9.5.orig/src/pcmanfm.c
+++ pcmanfm-0.9.5/src/pcmanfm.c
@@ -295,7 +295,7 @@ gboolean on_socket_event( GIOChannel* io
 void get_socket_name( char* buf, int len )
 {
     char* dpy = gdk_get_display();
-    g_snprintf( buf, len, "/tmp/.pcmanfm2-socket%s-%s", dpy, g_get_user_name() );
+    g_snprintf( buf, len, "/tmp/.pcmanfm-socket%s-%s", dpy, g_get_user_name() );
     g_free( dpy );
 }
 
--- pcmanfm-0.9.5.orig/src/Makefile.am
+++ pcmanfm-0.9.5/src/Makefile.am
@@ -1,13 +1,13 @@
 NULL=
 
 AM_CPPFLAGS = \
-	-DPACKAGE_DATA_DIR=\""$(datadir)/pcmanfm2"\" \
-	-DPACKAGE_UI_DIR=\""$(datadir)/pcmanfm2/ui"\" \
+	-DPACKAGE_DATA_DIR=\""$(datadir)/pcmanfm"\" \
+	-DPACKAGE_UI_DIR=\""$(datadir)/pcmanfm/ui"\" \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
 
-bin_PROGRAMS = pcmanfm2
+bin_PROGRAMS = pcmanfm
 
-pcmanfm2_SOURCES = \
+pcmanfm_SOURCES = \
 	pcmanfm.c pcmanfm.h \
 	app-config.c app-config.h \
 	main-win.c main-win.h \
@@ -22,7 +22,7 @@ EXTRA_DIST= \
 	desktop-ui.c \
 	$(NULL)
 
-pcmanfm2_CFLAGS = \
+pcmanfm_CFLAGS = \
 	$(XLIB_CFLAGS) \
 	$(GTK_CFLAGS) \
 	$(GIO_CFLAGS) \
@@ -34,7 +34,7 @@ pcmanfm2_CFLAGS = \
 	-Werror-implicit-function-declaration \
 	$(NULL)
 
-pcmanfm2_LDADD = \
+pcmanfm_LDADD = \
 	$(XLIB_LIBS) \
 	$(GTK_LIBS) \
 	$(GIO_LIBS) \
--- pcmanfm-0.9.5.orig/src/main-win.c
+++ pcmanfm-0.9.5/src/main-win.c
@@ -625,9 +625,9 @@ void on_open_as_root(GtkAction* act, FmM
         return;
     }
     if(strstr(app_config->su_cmd, "%s")) /* FIXME: need to rename to pcmanfm when we reach stable release. */
-        cmd = g_strdup_printf(app_config->su_cmd, "pcmanfm2 %U");
+        cmd = g_strdup_printf(app_config->su_cmd, "pcmanfm %U");
     else
-        cmd = g_strconcat(app_config->su_cmd, " ", "pcmanfm2 %U", NULL);
+        cmd = g_strconcat(app_config->su_cmd, " ", "pcmanfm %U", NULL);
     app = g_app_info_create_from_commandline(cmd, NULL, 0, NULL);
     g_free(cmd);
     if(app)
--- /dev/null
+++ pcmanfm-0.9.5/patches/series
@@ -0,0 +1 @@
+01_binary_rename.patch
